site/app/models/gradeable/Mark.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace app\models\gradeable;
+
+use app\libraries\Core;
+use app\models\AbstractModel;
+
+/**
+ * Class Mark
+ * @package app\models\gradeable
+ *
+ * @method getId();
+ * @method setId($id);
+ * @method getPoints();
+ * @method getNote();
+ * @method setNote($note);
+ * @method getOrder();
+ * @method setOrder($order);
+ * @method isPublish();
+ * @method setPublish($should_publish);
+ */
+class Mark extends AbstractModel
+{
+    /** @property @var int The course-wide unique numeric id of this mark */
+    protected $id = -1;
+    /** @property @var int The number of points this mark will add to the score (negative for deductions) */
+    protected $points = 0;
+    /** @property @var string The description of this mark (aka why a student would lose/gain these points) */
+    protected $note = "";
```

Comment:
Gradeables have `title`, Components have `title`, Mark should have `title` as well